pipeline/metadata/satellite.py

```diff
@@ -92,6 +92,46 @@ def get_blockpage_table_name(table_name: str, scan_type: str) -> str:
                             f'.{scan_type}_{SCAN_TYPE_BLOCKPAGE}')
 
 
+def _get_received_ips_with_roundtrip_id_and_date(row: Row) -> Iterable[Row]:
+  """Get all the individual received ips answers from a row.
+
+  Args:
+    row: dicts without metadata in the received ips lke:
+      {
+        'ip': '1.2.3.4'
+        'date': '2021-01-01'
+        'domain': 'ex.com'
+        'roundtrip_id: 'abc'
+        'received' : [{
+            'ip': '4.5.6.7'
+          }, {
+            'ip': '5.6.7.8'
+          },
+        ]
+      }
+
+  Yields individual recieved answers that also include the roundtrip and date
+    ex:
+      {
+        'ip': '4.5.6.7',
+        'date': '2021-01-01',
+        'roundtrip_id: 'abc'
+      }, {
+        'ip': '5.6.7.8',
+        'date': '2021-01-01',
+        'roundtrip_id: 'abc'
+      }
+  """
+  roundtrip_id = row['roundtrip_id']
+  date = row['date']
+
+  for answer in row['received']:
+    received_with_id = answer.copy()
```

Comment:
This is abusing the received type.
Why not put the entire received data in a field, instead of expanding it in the parent dict?

Reply:
Let's punt on this for now.